### PR TITLE
fix(web-components): added prop to prevent auto highlighting first op…

### PR DESCRIPTION
**Pull request by GCHQ-Developer-741**
…tion

this prop gives people the ability to submit terms not included in the options list, since it won't automatically populate the field with the first option

## Summary of the changes
In order to allow users to submit their own custom search terms, the automatic highlighting of the first option needed to be removed, so that the field doesn't get populated on submission of search. However, there are a few scenarios where this functionality is needed, for example on our guidance site, which functions more as a quick navigation tool and therefore relies on having the options highlighted to avoid 404 errors.

To add flexibility, I have added a new prop called `searchMode`, which is `navigation` by default but users can set this to `query` if they want the search bar to function more traditionally.

A new ticket has been raised in design (https://github.com/mi6/ic-ui-kit/issues/897) to add some visual changes to the search bar when this prop is enabled, to make clear the difference in behaviour when this prop is used.

## Related issue
#798 

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

**Review by DBD324 — CHANGES_REQUESTED** [review]
Think we need to review the terminology here. The issue isn't as simple as just disabling auto highlighting. It actually creates a totally different mode of search interaction. Think the work is sound, we just need to be really careful about the prop naming and the prop description as it changes the search mode drastically. 

Perhaps we could use something like:
- goToSearchResult: false

or conversely
-searchForInputTerm: true. 

I would like the team to think about wording this so that its impact and function is understandable.

**Review by jd239 — DISMISSED** [review]
Nice work. I will approve as the code update is good. :)

File: packages/web-components/src/components/ic-search-bar/ic-search-bar.tsx
Comment (GCHQ-Developer-847) [inline packages/web-components/src/components/ic-search-bar/ic-search-bar.tsx:1]: Tried adding the prop set to "query" on the "With Options" story - if I type something in and then start deleting the text I've typed in, the first option gets highlighted

**Review by GCHQ-Developer-847 — CHANGES_REQUESTED** [review]

**Comment by GCHQ-Developer-847** [comment]
Just noticed another thing actually - on the React "With options" story, I've tried setting searchMode to "query" and then added `onIcSubmitSearch={(ev) => console.log(ev.detail.value)}`. When I type something in the input box and press enter, nothing gets logged to the console. 

So when someone sets the search mode to "query", should it actually allow them to submit it i.e. should it emit onIcSubmitSearch with the value they've typed in when they press enter? That means they can either submit the value of one of their options, or the custom value they've typed in (a bit like how it works in the "Default" story - whatever you've typed in gets submitted)

**Comment by GCHQ-Developer-847** [comment]
> Just noticed another thing actually - on the React "With options" story, I've tried setting searchMode to "query" and then added `onIcSubmitSearch={(ev) => console.log(ev.detail.value)}`. When I type something in the input box and press enter, nothing gets logged to the console.
> 
> So when someone sets the search mode to "query", should it actually allow them to submit it i.e. should it emit onIcSubmitSearch with the value they've typed in when they press enter? That means they can either submit the value of one of their options, or the custom value they've typed in (a bit like how it works in the "Default" story - whatever you've typed in gets submitted)

^ Actually this does work - it does get submitted, but only when there options. E.g. on the "With options" story, I can submit "ca" or "am" for example, as that brings up options that match. But when I type in a random string which doesn't match any option (when it says "No results found"), I can't submit it

**Review by GCHQ-Developer-847 — APPROVED** [review]
Approved (we have spoken about my previous comment and I am happy that it is all working as it should - was possibly fixed by another change)

**Review by DBD324 — APPROVED** [review]
Terminology seems much more understandable now, thank you.

**Pull request merged**